### PR TITLE
Issue/11114 avoid tracking signup as error

### LINKED ISSUE #11114: [REST API] Add support for WordPress.com account creation as part of Jetpack Setup
Currently, in the Jetpack Setup flow, we don't support creating new accounts, which is counter-intuitive, as users who are installing Jetpack for the first time will most likely need to create a new account.

Using tracks, we can confirm this: currently, the login step is one of the biggest drops in the Jetpack Setup, as only 22% are able to finish it. From the failures, we can see that `UNKOWN_USER` is responsible for the biggest part of the cases.

###### iOS GH issue: https://github.com/woocommerce/woocommerce-ios/issues/12310

```[tasklist]
### Tasks
- [ ] https://github.com/woocommerce/woocommerce-android/issues/12918
- [ ] https://github.com/woocommerce/woocommerce-android/issues/12921
- [ ] https://github.com/woocommerce/woocommerce-android/issues/12944
```

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #11114 

### Description

Small change on how we track failures during login step in Jetpack installation flow. 
Full discussion here: p1732113936622179/1731431785.549369-slack-C03L1NF1EA3
TLDR: We won't track `signup` case as `woocommerceandroid_jetpack_setup_login_flow, Properties: {"step":"email_address","failure":"UNKNOWN_USER", ...` event anymore. 

Additionally this PR adds a small update on `username_not_registered_wpcom` removing the `Hmm` part and just leaving it as: `We can\'t find a WordPress.com account connected to this username. You can enter an email to create a new account.`

Screenshot 2024-11-21 at 13 24 34

### Testing information

1. Log into a self hosted site that is not Jetpack connected
2. Click on Jetpack banner and enter an invalid `username`
3. Check that the error message is shown as expected and check that `woocommerceandroid_jetpack_setup_login_flow, Properties: {"step":"email_address","failure":"UNKNOWN_USER"` is tracked in the Logcat
4. Erase the invalid username and enter a unknown WP.com email
5. Check that no event for `failure` is tracked. 

### The tests that have been performed

The above

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/244">21.2</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by hichamboushaba — APPROVED** [review]
Looks good, thanks @JorgeMucientes
